hedy.py

```diff
@@ -1932,31 +1939,56 @@ def play(self, meta, args):
     def assign(self, meta, args):
         variable_name = args[0]
         value = args[1]
-
-        if self.is_random(value) or self.is_list(value):
-            exception = self.make_index_error_check_if_list([value])
-            return exception + variable_name + " = " + value + self.add_debug_breakpoint()
+        if not self.microbit:
+            if self.is_random(value) or self.is_list(value):
```

Comment:
I like that the new microbit code path is non-intrusive. The code that needs to be different is copied in an if-microbit-else statement. My only concern is that the current code of the assign function needs to be refactored because it contains useless checks and repetition. I'd update the original code of the function in the following way: 
```
            exception = self.make_index_error_check_if_list([value])
            if self.is_variable(value, meta.line):
                # if the assigned value is a variable, this is a reassign
                value = self.process_variable(value, meta.line)
            else:
                # if it is not a variable and contains single quotes, escape them
                value = f"'{process_characters_needing_escape(value)}'"
            return exception + variable_name + " = " + value + self.add_debug_breakpoint()
```
This would also simplify the microbit part, I think. However, if this refactoring is too much for the current task at hand, please close this comment and I will add a note to refactor this piece of code at a later stage.

Reply:
After disusing this issue and choosing to have a separate transpiler for Micro:bit this functions no longer needs to be modified for Micro:bit.